### PR TITLE
Implement logging class to write logs in txt file

### PR DESCRIPTION
**Pull request by PureUsagi**
Added a new class called MainLogger that will handle all logging funtionality of BinBash.
Every class must declare a MainLogger object  if they are to write logs to the log file.

### CODE REVIEW

File: logs/logs.txt
Comment (YHWong20) [inline logs/logs.txt:1]: Don't commit this file, add it to `.gitignore`.

File: src/main/java/seedu/binbash/logger/MainLogger.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+import java.io.IOException;
+import java.util.logging.FileHandler;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.logging.SimpleFormatter;
+
+public class MainLogger {
+    private static FileHandler fileHandler;
+    private static final String logDirectoryPath = "./logs/";
+    private static final String logFileName = "logs.txt";
```
Comment (YHWong20) [inline src/main/java/seedu/binbash/logger/MainLogger.java:15]: This is not an urgent/crucial fix, but we can probably make use of the `java.nio` package in the future to handle file paths.

File: src/main/java/seedu/binbash/logger/MainLogger.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+                throw new BinBashException("Could not create log file.");
+            }
+
+            if(!wasLogFileCreated) {
+                throw new BinBashException("Could not create log file.");
+            }
+        }
+
+        assert logDirectory.exists() : "Log directory should already exist / have been created";
+        assert logFile.exists() : "Log file (logs.txt) should already exists / have been created";
```
Comment (YHWong20) [inline src/main/java/seedu/binbash/logger/MainLogger.java:49]: Good usage of assertion here.

File: logs/logs.txt
Comment (PureUsagi) [reply logs/logs.txt:1]: Okayyy.

File: src/main/java/seedu/binbash/logger/MainLogger.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    private void setFileHandler() throws BinBashException {
+        try {
+            fileHandler = new FileHandler(logDirectoryPath + logFileName);
+        } catch (IOException e) {
+            throw new BinBashException("Could not create file handler! Unable to generate logs!");
+        }
+
+        fileHandler.setFormatter(new SimpleFormatter());
+        fileLogger.setUseParentHandlers(false);
+        fileLogger.addHandler((fileHandler));
```
[the remark this answers is not in the document]
Comment (PureUsagi) [reply src/main/java/seedu/binbash/logger/MainLogger.java:82]: Oh, that is unintentional, good spot!

File: src/main/java/seedu/binbash/logger/MainLogger.java (hunk 1 above)
Comment (PureUsagi) [reply src/main/java/seedu/binbash/logger/MainLogger.java:15]: Okayyy, we will leave it for now

**Review by YHWong20 — APPROVED** [review]
Kindly help to merge after rectifying the merge conflicts. Thanks!

**Pull request merged**